Controllers/HomeController.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Zoltu.Bags.Client.Controllers
+{
+	public class HomeController : Controller
+	{
+		[Route("")]
+		[Route("app")]
+		[Route("app/{path}")]

```

Comment:
This needs to be a catch-all route: `app/{*path}`

There are a ton of different things that can go after the `/app/`, product is only one small portion of it.  The logic for parsing out the path will eventually occur inside the `App` action.